tutorials/persistance-comparison-series/PlayerPrefs/player_prefs.md

```diff
@@ -0,0 +1,436 @@
+
+One possible approach would be to use the fact that `PlayerPrefs` can hold a `string` and save a `JSON` in there.
+
+First we need to figure out how to actually transform our data into JSON. The .NET framework as well as the `UnityEngine` framework offer a JSON serialize and deserializer to do this job for us. Both behave very similar but we will use Unity's own [`JsonUtility`](https://docs.unity3d.com/ScriptReference/JsonUtility.html) which [performs better in Unity than other similar JSON solution](https://docs.unity3d.com/Manual/JSONSerialization.html).
+
+To transform data to JSON we first need to create a container object. This has [some restriction](https://docs.unity3d.com/ScriptReference/JsonUtility.ToJson.html):
+
+> Internally, this method uses the Unity serializer; therefore the object you pass in must be supported by the serializer: it must be a MonoBehaviour, ScriptableObject, or plain class/struct with the Serializable attribute applied. The types of fields that you want to be included must be supported by the serializer; unsupported fields will be ignored, as will private fields, static fields, and fields with the NonSerialized attribute applied.
+
+In our case, since we are only saving simple data types (int) for now that's fine. We can define a new class (1) and call it `HitCount`:
+
+```cs
+// 1
+private class HitCount
+{
+    public int Unmodified;
+    public int Shift;
+    public int Control;
+}
+```
+
+We will keep the Unity editor outlets the same (2):
+
+```cs
+// 2
+[SerializeField] private int hitCountUnmodified = 0;
+[SerializeField] private int hitCountShift = 0;
+[SerializeField] private int hitCountControl = 0;
+```
+
+All those will eventually be saved into the same `PlayerPrefs` field which means we only need one key (3):
+
+```cs
+// 3
+private readonly string hitCountKey = "HitCountKeyJson";
+```
+
+As before, the `keyPressed` will indicate which modifier was used:
+
+```cs
+// 4
+private KeyCode keyPressed = default;
+```
+
+In `Start()` we then need to read the JSON. As before, we check if the `PlayerPrefs` key exists (5) and then read the data, this time using `GetString()` (as opposed to `GetInt()` before).
+
+Transforming this JSON into the actual object is then done using `JsonUtility.FromJson()` (6) which takes the string as an argument. It's a generic function and we need to provide the information about which object this JSON is supposed to be representing, in this case `HitCount`.
+
+If the JSON could be read and transformed successfully we can set the hit count fields (7) to their three values.
+
+```cs
+private void Start()
+{
+    // 5
+    // Check if the key exists. If not, we never saved to it.
+    if (PlayerPrefs.HasKey(HitCountKey))
+    {
+        // 6
+        string jsonString = PlayerPrefs.GetString(HitCountKey);
```

Comment:
These can be `var`-s, though I don't know if Unity devs prefer the explicit type. I do find it a bit repetitive `string jsonString = ..GetString(...)` - it's fairly obvious it's a string 😆 I would probably write it as:

```csharp
var json = PlayerPrefs.GetString(...);
var hitCount = JsonUtility.FromJson<HitCount>(...);
```

but it's a personal preference, so fine to keep it as is.

Reply:
Not sure about Unity developers. In this case it's definitely me just being used to explicit types from other languages. I'll never get rid of that habit I assume. :D
578743f